typed-racket-lib/typed-racket/types/prefab.rkt

```diff
@@ -0,0 +1,127 @@
+#lang racket/base
+
+;; Utilities for dealing with prefab struct types
+
+(require "../utils/utils.rkt"
+         (contract-req)
+         racket/list
+         racket/match)
+
+(provide/cond-contract [normalize-prefab-key
```

Comment:
This code should all go somewhere in core Racket -- see discussion w/ @mflatt elsewhere. But that's fine to do later, once we figure out what we want.